fdbserver/SimulatedCluster.actor.cpp

```diff
@@ -1579,12 +1581,60 @@ void SimulationConfig::setDatacenters(const TestConfig& testConfig) {
 	}
 }
 
+void SimulationConfig::setTenantMode(const TestConfig& testConfig) {
+	TenantMode tenantMode = TenantMode::DISABLED;
+	if (testConfig.tenantModes.size() > 0) {
+		tenantMode = TenantMode::fromString(deterministicRandom()->randomChoice(testConfig.tenantModes));
+	} else if (testConfig.allowDefaultTenant && deterministicRandom()->coinflip()) {
+		tenantMode = deterministicRandom()->random01() < 0.9 ? TenantMode::REQUIRED : TenantMode::OPTIONAL_TENANT;
+	} else if (deterministicRandom()->coinflip()) {
+		tenantMode = TenantMode::OPTIONAL_TENANT;
+	}
+	set_config("tenant_mode=" + tenantMode.toString());
+}
+
+void SimulationConfig::setEncryptionAtRestMode(const TestConfig& testConfig) {
+	EncryptionAtRestMode encryptionMode = EncryptionAtRestMode::DISABLED;
+	// Only Redwood support encryption. Disable encryption if non-Redwood storage engine is explicitly specified.
+	bool disableEncryption = testConfig.disableEncryption ||
+	                         (testConfig.storageEngineType.present() && testConfig.storageEngineType.get() != 3);
+	// TODO: Remove check on the ENABLE_ENCRYPTION knob once the EKP can start using the db config
+	if (!disableEncryption && (SERVER_KNOBS->ENABLE_ENCRYPTION || !testConfig.encryptModes.empty())) {
+		TenantMode tenantMode = db.tenantMode;
+		if (!testConfig.encryptModes.empty()) {
+			std::vector<EncryptionAtRestMode> validEncryptModes;
+			// Get the subset of valid encrypt modes given the tenant mode
+			for (int i = 0; i < testConfig.encryptModes.size(); i++) {
+				EncryptionAtRestMode encryptMode = EncryptionAtRestMode::fromString(testConfig.encryptModes.at(i));
+				if (encryptMode != EncryptionAtRestMode::DOMAIN_AWARE || tenantMode == TenantMode::REQUIRED) {
+					validEncryptModes.push_back(encryptMode);
+				}
+			}
+			if (validEncryptModes.size() > 0) {
+				encryptionMode = deterministicRandom()->randomChoice(validEncryptModes);
+			}
+		} else {
+			// TODO: These cases should only trigger with probability (BUGGIFY) once the server knob is removed
+			if (tenantMode == TenantMode::DISABLED || tenantMode == TenantMode::OPTIONAL_TENANT || BUGGIFY) {
```

Comment:
Was reading over this and realized that in this case we never set encryptMode to disabled (because it will either be cluster aware or domain aware always)

Reply:
lol, I was just copy-pasting existing code and didn't look at it carefully. will update it to test with encryption disabled randomly.

Reply:
Actually before we remove the ENABLE_ENCRYPTION knob this is the correct behavior, as the knob already gets randomization. I will update the logic here when I remove the knob.